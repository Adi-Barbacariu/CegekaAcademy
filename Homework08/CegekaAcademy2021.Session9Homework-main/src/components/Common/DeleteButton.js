import React, { useState } from "react";
import { Icon, Button, Confirm } from "semantic-ui-react";
import { useDispatch } from "react-redux";
import { deletePhoto } from "../../features/photosSlice.js";
import { deleteAlbum } from "../../features/albumsSlice";

function DeleteButton(props) {
  const [deleteConfirmOpen, setDeleteConfirmOpen] = useState(false);

  const { index, objectName, itemType } = props;
  const dispatch = useDispatch();

  const onDeleteItem = () => {
    if (itemType == "photo") {
      dispatch(deletePhoto(index));
    }

    if (itemType == "album") {
      dispatch(deleteAlbum(index));
    }

    setDeleteConfirmOpen(false);
  };

  const handleDelete = () => {
    setDeleteConfirmOpen(true);
  };

  const onCancelConfirm = () => {
    setDeleteConfirmOpen(false);
  };

  return (
    <Button basic color="red" onClick={() => handleDelete(index)}>
      <Icon name="trash" />
      Delete
      {deleteConfirmOpen && (
        <Confirm
          open={deleteConfirmOpen}
          content={`Are you sure you want to delete '${objectName}'?`}
          cancelButton="No"
          confirmButton="Yes, delete it!"
          onCancel={() => onCancelConfirm()}
          onConfirm={onDeleteItem}
        />
      )}
    </Button>
  );
}

export default DeleteButton;
