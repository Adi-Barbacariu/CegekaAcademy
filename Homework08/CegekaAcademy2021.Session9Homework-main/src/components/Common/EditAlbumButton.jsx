import React from "react";
import { Button, Header, Modal, Icon, Input } from "semantic-ui-react";

function EditAlbumButton({
  index,
  name,
  description,
  tags,
  photos,
  editAlbum,
}) {
  const [open, setOpen] = React.useState(false);
  const [albumData, setAlbumData] = React.useState({
    name,
    description,
    tags,
    photos,
  });

  const handleChange = (e) => {
    const value = e.target.value;

    setAlbumData({ ...albumData, [e.target.name]: value });
  };

  const handleCancel = (e) => {
    setAlbumData({ name, description });
    setOpen(false);
  };

  const handleEdit = (e) => {
    editAlbum(index, { ...albumData });
    setOpen(false);
  };

  return (
    <Button basic color="green" onClick={() => setOpen(true)}>
      <Icon name="edit" />
      Edit
      {open && (
        <Modal
          onClose={() => setOpen(false)}
          onOpen={() => setOpen(true)}
          open={open}
        >
          <Modal.Header>Edit Album</Modal.Header>
          <Modal.Content image>
            <Modal.Description>
              <Header as="h3">Name</Header>
              <Input
                fluid
                defaultValue={albumData.name}
                name="name"
                onChange={handleChange}
              />
              <Header as="h3">Description</Header>
              <Input
                fluid
                defaultValue={albumData.description}
                name="description"
                onChange={handleChange}
              />
            </Modal.Description>
          </Modal.Content>
          <Modal.Actions>
            <Button color="black" onClick={handleCancel}>
              Cancel
            </Button>
            <Button
              content="Done"
              labelPosition="right"
              icon="checkmark"
              onClick={handleEdit}
              positive
            />
          </Modal.Actions>
        </Modal>
      )}
    </Button>
  );
}

export default EditAlbumButton;
