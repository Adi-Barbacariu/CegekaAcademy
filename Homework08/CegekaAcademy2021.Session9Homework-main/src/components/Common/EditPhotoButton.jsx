import React from "react";
import { Button, Header, Modal, Icon, Input } from "semantic-ui-react";
import { useDispatch } from "react-redux";
import { editPhoto } from "../../features/photosSlice.js";

function EditPhotoButton({ index, title, description, url }) {
  const [open, setOpen] = React.useState(false);
  const [photoData, setPhotoData] = React.useState({ title, description, url });

  const handleChange = (e) => {
    const value = e.target.value;

    setPhotoData({ ...photoData, [e.target.name]: value });
  };

  const handleCancel = (e) => {
    setPhotoData({ title, description, url });
    setOpen(false);
  };

  const dispatch = useDispatch();

  const onEditPhoto = () => {
    dispatch(editPhoto({ index, photo: { ...photoData } }));
    setOpen(false);
  };

  return (
    <Button basic color="green" onClick={() => setOpen(true)}>
      <Icon name="edit" />
      Edit
      {open && (
        <Modal
          onClose={() => setOpen(false)}
          onOpen={() => setOpen(true)}
          open={open}
        >
          <Modal.Header>Edit Photo</Modal.Header>
          <Modal.Content image>
            <Modal.Description>
              <Header as="h3">Title</Header>
              <Input
                fluid
                defaultValue={photoData.title}
                name="title"
                onChange={handleChange}
              />
              <Header as="h3">Description</Header>
              <Input
                fluid
                defaultValue={photoData.description}
                name="description"
                onChange={handleChange}
              />
              <Header as="h3">URL</Header>
              <Input
                fluid
                defaultValue={photoData.url}
                name="url"
                onChange={handleChange}
              />
            </Modal.Description>
          </Modal.Content>
          <Modal.Actions>
            <Button color="black" onClick={handleCancel}>
              Cancel
            </Button>
            <Button
              content="Done"
              labelPosition="right"
              icon="checkmark"
              onClick={onEditPhoto}
              positive
            />
          </Modal.Actions>
        </Modal>
      )}
    </Button>
  );
}

export default EditPhotoButton;
