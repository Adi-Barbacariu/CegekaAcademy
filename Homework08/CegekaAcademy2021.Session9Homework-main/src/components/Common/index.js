export { default as WithLightbox } from "./WithLightbox";
export { default as DeleteButton } from "./DeleteButton";
export { default as EditPhotoButton } from "./EditPhotoButton.jsx";
export { default as EditAlbumButton } from "./EditAlbumButton.jsx";
