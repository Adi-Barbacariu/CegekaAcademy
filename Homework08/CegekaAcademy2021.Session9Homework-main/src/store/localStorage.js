import * as api from "../api";

export const loadPhotos = () => {
  const localPhotos = localStorage.getItem("photos");

  if (localPhotos) {
    return JSON.parse(localPhotos);
  }

  return api.getPhotos();
};

export const loadAlbums = () => {
  const localAlbums = localStorage.getItem("albums");

  if (localAlbums) {
    return JSON.parse(localAlbums);
  }

  return api.getAlbums();
};

export const saveState = (itemName, state) => {
  localStorage.setItem(itemName, JSON.stringify(state));
};
