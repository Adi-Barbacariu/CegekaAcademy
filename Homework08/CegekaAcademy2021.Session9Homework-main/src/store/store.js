import { configureStore } from "@reduxjs/toolkit";
import photosReducer from "../features/photosSlice";
import albumsReducer from "../features/albumsSlice";

export default configureStore({
  reducer: {
    photos: photosReducer,
    albums: albumsReducer,
  },
});
