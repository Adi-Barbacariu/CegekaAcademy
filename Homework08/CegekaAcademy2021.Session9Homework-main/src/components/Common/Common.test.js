/* global describe, it, expect, renderer */
import React from "react";
import { DeleteButton } from "../Common";
import { Provider } from "react-redux";
import store from "../../store/store";

describe("Common", () => {
  it("Delete button snapshot test", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DeleteButton
            index="index"
            objectName="My object name"
            itemType="photo"
          />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
