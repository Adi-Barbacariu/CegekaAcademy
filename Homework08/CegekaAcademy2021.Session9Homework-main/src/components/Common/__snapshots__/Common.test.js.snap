// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Delete button snapshot test 1`] = `
<button
  className="ui red basic button"
  disabled={undefined}
  onClick={[Function]}
  role="button"
  tabIndex={undefined}
>
  <i
    aria-hidden="true"
    className="trash icon"
  />
  Delete
</button>
`;
